### PR TITLE
Introduce Animation editor

### PR DESCRIPTION
**Pull request by ZacharieALES**
Here is the code for the animation and the start of the code for the editor.

I expected to clean many things after finishing a first version of the editor (which is still not the case), so brace yourselves if you have a look in the code.

To make it work quickly I modified DrillPage so that it always display an animation (or the editor depending on the case).

### CODE REVIEW

File: src/Components/animation/AnimationDE.js [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+
+import DisplayedElement from './DisplayedElement';
+import DrillMenageATrois from './DrillMenageATrois';
+import DrillSquare from './DrillSquare';
+
+/* TODO:
+ * - when moving to a given step stepId, put back all displayed elements to their position prior stepId
+ * - when restarting the complete animation, if the elements are not in their initial positions, add an empty animation to move the elements to their initial position before starting the animation
+ */
+class AnimationDE extends React.Component {
```
[the remark this answers is not in the document]
Comment (ZacharieALES) [reply src/Components/animation/AnimationDE.js:12]: An old file that should be removed

File: src/Components/animation/Test.js [hunk 2]
```diff
@@ -0,0 +1,222 @@
+import React from 'react';
+import { StyleSheet, Easing, Animated, View, PanResponder } from 'react-native';
+
+/** An element displayed in a drill animation */
+class Test extends React.Component {
```
[the remark this answers is not in the document]
Comment (ZacharieALES) [reply src/Components/animation/Test.js:9]: They should be merged in the end. Test represents an element displayed at the bottom of the editor which enables to add this element to the drill.

DisplayedElement represents an element in a drill.

So in the editor, when you drag an element Test at a position (x, y) in the animation area :
- a DisplayedElement is created at position (x, y) ;
- the Test which was dragged is moved to its initial position.

I found it easier to use 2 classes when creating the editor.

File: src/Components/animation/ElementAdderBar.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import { StyleSheet, Animated, View } from 'react-native';
+
+import DisplayedDraggableElement from './DisplayedDraggableElement';
+
+class ElementAdderBar extends React.Component {
```
[the remark this answers is not in the document]
Comment (ZacharieALES) [reply src/Components/animation/ElementAdderBar.js:6]: Yes

File: src/Components/animation/Test.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (sallesma) [reply src/Components/animation/Test.js:9]: Aaaaah, ok ! :)

[the remark this answers is not in the document]
Comment (sallesma) [reply src/Components/animation/Test.js:9]: Maybe rename it `DraggableDisplayedElement` ?

File: src/Components/DrillIllustration.js [hunk 4]
```diff
@@ -90,7 +90,7 @@ const DrillIllustration = props => {
   const displayAnimation = ({ illustrationSource, instruction, title }) => {
     return (
       <>
-        <Animation readonly widthRatio={1} heightRatio={1 / 2} animation={illustrationSource} />
```
Comment (sallesma) [inline src/Components/DrillIllustration.js:93]: I considered an Animation is readonly by default, and must be tagged as editable to be used in the editor

**Pull request merged**